### DISCUSSION TITLE
Stuck on loading after v4 update? Please read.

### DISCUSSION THREAD

**Discussion by bluecaret** [bluecaret/carettab#345] [! News and Announcements]
**The following should only apply to users updating from v3 to v4.**

It seems that the v4 update has caused some users to experience a stuck loading screen after the update. I'm looking into this issue now but due to Chrome's update policy, when I do find a fix, they won't push the update for a day or two anyway.

An immediate fix for this is unfortunately to uninstall, then reinstall, which will reset all of your settings. However, this update did unfortunately require a reset of most settings anyway due to a big change in how widgets are configured, which normally would have been explained on the next screen.

I'm very sorry for the trouble this has caused, this was a big update that I was very excited to release, but it seems I've botched the update. I hope you will all have patience with the transition and stick with it. This update does unlock the door for huge improvements in the future!

EDIT: Version 4.0.4 is releasing soon which resolves this issue. If you continue to have issues, be sure you are on this version or later.

**EDIT 2: If you are still experiencing the stuck loading screen even after version 4.0.4, you may need to uninstall and reinstall the extension. I've done all I can to try to fix the bug but it seems to be once you hit that, it's stuck until a reset.**

**EDIT 3: I just pushed out version 4.1.0 which has what I believe is the final fix for this loading issue. Please keep a look out for this update in the next couple days. Very sorry for the amount of time it has taken to get to this point.**

**Comment by bluecaret** [comment]
Just pushed a fix out. Chrome should get it published hopefully in a day or two. If you are on the stuck loading screen, wait till the next update gets out and the extension should be able to continue with the setup of v4. Thank you for your patience.

**Reply by cikloartstudio** [reply]
Don't worry, it's really the first time this happens to me, (just for reporting) I'm using Linux Ubuntu Studio 22.04 LTS. So thank you for your willingness and great speed to solve the update!!!

**Comment by j-freddy** [comment]
[Chrome] My experience is that it stucks on load after auto-migrating from v3 to v4, but restarting Chrome fixes the issue. (on OS X and Windows)

**Reply by bluecaret** [reply]
Thank you. version 4.0.2 went out late yesterday and this should have resolved the issue. I'm glad it is working for you again.

**Comment by romomartin** [comment]
Still experiencing the infinite loading screen issue with this config:

MacOS
Chrome Version 115.0.5790.114 (Official Build) (arm64) 
CaretTab version 4.0.3

Console error:
<img width="555" alt="error" src="https://github.com/bluecaret/carettab/assets/72378221/ce149125-5a4e-4fd0-a8d7-22f797768f97">

**Reply by bluecaret** [reply]
Thank you very much for letting me know. It seems I missed a bug, the screenshot is very helpful. I haven't been able to reproduce this on my machine so it is a bit of trial and error. I appreciate your patience.

**Reply by romomartin** [reply]
Update worked fine this time with 4.0.4, only downside is that previous configuration is lost. 

Anyway, thank you for the good work with the extension. New design and functionalities look awesome!

**Reply by bluecaret** [reply]
Thank goodness!! I'm going to have nightmares about this update forever 😥

I'm glad you are all up and running again and that you are enjoying the new features! Cheers!

**Comment by bluecaret** [comment]
I apologize everyone again. I was finally able to reproduce the error which allowed me to fix it for sure this time! Just pushed the update to chrome, they approved very quickly with the last one so hoping they do the same here again. Keep a look out for **version 4.0.4**

**Comment by frank42hh** [comment]
HI!
It seems that v4 completely fsck'ed up my CaretTab settings. Everything's gone. The last settings backup from pre-v4 can not be loaded. Is this intended?

**Reply by bluecaret** [reply]
I apologize for the trouble. Due to a change in how widgets are configured, clearing the settings was required as a migration was impossible for most options, this was needed for changes to how widgets are positioned as well as more control over some existing settings. Some items such as quick links were supposed to migrate but there was a bug in that. Despite the issues I hope you continue to give the update a chance, I think you will find that there are many new improvements that will hopefully dull the pain from the update.

**Reply by frank42hh** [reply]
Thanks for the response. Of course I give it a chance, since it's a very helpful extension. It's just... I had a very nice layout and I tried for a long time yesterday to recreate my landing page with the new system but failed so far. It sure has several new and very interesting features, but I'm a bit overwhelmed with it. One thing for example is that I'm missing that 7-segment-display font, which I really enjoyed.

**Reply by bluecaret** [reply]
Posted this in the wrong spot originally, so here is my earlier reply: 

I can understand, and am very sorry for the frustration. Thank you for pointing out the 7-segment font, I overlooked the removal of that. It is not available from Google Fonts which is why it is missing, but it is an important font for a lot of people so I'll have to provide the option another way. I'll get that added back. Thank you!

Also as an update since I originally wrote that, I've added a few 7 segment fonts and have included them in an update that was just sent to Chrome today, 4.1.0. Keep a look out!

**Comment by Poetry-PoetPatriot-com** [comment]
Will the new version keep my bookmarks?
FYI - still stuck at this time.

**Comment by bluecaret** [comment]
To all still experiencing the issue. I believe I have a final fix in place for this now and just pushed an update out today. It should get rolled out in a couple days. Keep a look out for v4.1.0. Thank you.
